### PR TITLE
refactor: adopt variable

### PR DESCRIPTION
**Pull request by kellyjosephprice**
[][demo] |
:---:|

## 🧰 Changes

Adopts the `@readme/variable` package. From now on, you'll be able to import the `Variable`, `VARIABLE_REGEX`, and `SelectedAppContext` from `@readme/markdown`.

Note that `@readme/variable`'s old default export was the `Variable` component with the requisite context consumers. At the same time, it was exporting the `Variable` component as the named export: `Variable`. `@readme/markdown` will only be exporting the component wrapped with the consumers.

## TODO

There's a bit of a chicken and the egg type problem. This repo is depending on `@readme/syntax-highlighter` which is depending on `@readme/variable`. So we'll have to push an updated `@readme/markdown` first, before we can update `@readme/syntax-highlighter` to use `@readme/markdown`.

## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-200.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -32,7 +32,9 @@ const rehypeReact = require('rehype-react');
  */
 const BaseUrlContext = require('./contexts/BaseUrl');
 
-const Variable = require('@readme/variable');
+const Variable = require('./variable');
+
+export const SelectedAppContext = Variable.SelectedAppContext;
```
Comment (erunion) [inline index.js:37]: Can you also retain the `Variable` and `VariablesContext` exports that `@readme/variable` has? We use those in `@readme/syntax-highlighter`: https://github.com/readmeio/syntax-highlighter/blob/main/src/codeMirror/index.jsx#L4

Though I guess that library can load them with `@readme/markdown/variable`? Up to you.

Comment (kellyjosephprice) [reply index.js:37]: Oh, oops. I meant to do that.

**Review by erunion — APPROVED** [review]
This all looks good to me! If you tag a new release of Markdown I can get a new build of `@readme/syntax-highlighter` together with it pretty quickly for you.

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.28.0-next.1</b>](/readmeio/markdown/releases/tag/v6.28.0-next.1)
<br>

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.28.0</b>](/readmeio/markdown/releases/tag/v6.28.0)
<br>

**Pull request merged**